feat: Support Kicks Gifted event

Doc94 opened the pull request:
@TeksuSiK not sure if i make this correctly but the PR give support to the kicks gifted event.
ref: https://docs.kick.com/events/event-types#kicks-gifted

Doc94 committed:
feat: Support Kicks Gifted event

Took 14 minutes


Took 2 minutes

Took 2 minutes

TeksuSiK commented on src/main/java/pl/teksusik/kick4j/events/type/Gift.java:
```diff
@@ -0,0 +1,45 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Gift {
+    private final int amount;
```
Use boxed type instead of primitive one

TeksuSiK commented on src/main/java/pl/teksusik/kick4j/events/type/Gift.java:
```diff
@@ -0,0 +1,45 @@
+    private final String name;
+    private final String type;
+    private final String tier;
+    private final String message;
```
According to KickDev documentation this class is missing `pinned_time_seconds` field

TeksuSiK commented on src/test/java/pl/teksusik/kick4j/events/KicksGiftedEventTest.java:
```diff
@@ -0,0 +1,79 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class KicksGiftedEventTest extends KickEventDispatcherTest {
+    private static final String PAYLOAD = """
```
Please update payload to match that one provided in [KickDev documentation](https://docs.kick.com/events/event-types#kicks-gifted)

TeksuSiK reviewed (changes requested):
Hi, thanks for your contribution! Please update code according to review 😊

Doc94 committed:
feat: changes for Gift

- Change amount to boxed type
- Add field for pinnedTimeSeconds

Took 6 minutes


Took 14 seconds

Doc94 committed:
feat: Change Tests for Gifts using the updated docs

Took 18 seconds

Doc94 committed:
feat: minor change

Took 2 minutes

Doc94 commented:
@TeksuSiK rebased

Doc94 commented:
This was updated @TeksuSiK
